tests/run/static_method_inheritance.pyx

```diff
@@ -0,0 +1,36 @@
+# mode: run
+
+cdef class A:
+    @staticmethod
+    def moo():
+        return "moo"
+
+cdef class B(A):
+    @staticmethod
+    def moo():
+        return "boo"
```

Comment:
Let's make the test easier to understand and more helpful for debugging in failure cases.
```suggestion
cdef class A:
    @staticmethod
    def moo():
        return "mooA"

cdef class B(A):
    @staticmethod
    def moo():
        return "mooB"
```